applications/ShallowWaterApplication/custom_python/add_custom_utilities_to_python.cpp

```diff
@@ -57,6 +57,9 @@ namespace Python
         .def("ComputeHeightFromFreeSurface", &ShallowWaterUtilities::ComputeHeightFromFreeSurface)
         .def("ComputeVelocity", &ShallowWaterUtilities::ComputeVelocity)
         .def("ComputeMomentum", &ShallowWaterUtilities::ComputeMomentum)
+        .def("UpdatePrimitiveVariables", py::overload_cast<ModelPart&>(&ShallowWaterUtilities::UpdatePrimitiveVariables))
```

Comment:
I am curious, why do you need `py::overload_cast` ?

Reply:
The method is overloaded at c++ level, so I need to tell pybind11 which of the interfaces I am exposing.

Reply:
I think there is something similar in other python interfaces, for example add_model_part_to_python.cpp. But there are defined some auxiliary functions instead of using `py:overload_cast`

Reply:
Which would be the best way to expose overloaded methods?

Reply:
If there are some suggestions, I can make another PR

Reply:
I didn't know abt `py:overload_cast`!
If it does what I think it does then I think it is the best (since pybind-native) solution, much better than the fct-pointers or the aux-fcts
I will try this asap, hopefully we can get rid of lots of code :D

@jcotela @roigcarlo did you know abt this?

Reply:
it works with c++14, see [overloaded-methods](https://pybind11.readthedocs.io/en/stable/classes.html#overloaded-methods)

Reply:
Nooooooooooooo :cry: 

so your app will not compile with c++11 any more ...

@KratosMultiphysics/technical-committee another motivation to move to 14 :D

I will keep this in mind :)

Reply:
that's the advantage of research and small app's :smiley:

Reply:
until you need to compile on a cluster that barely supports c++11 :see_no_evil:

Reply:
you're right...